Code/LauncherUnified/Launcher.cpp

```diff
@@ -419,6 +418,13 @@ namespace O3DELauncher
         settingsRegistry->Set(LauncherTypeTag, launcherType);
         // Also add the launcher type as a specialization as well
         AZ::SettingsRegistryMergeUtils::MergeSettingsToRegistry_AddSpecialization(*settingsRegistry, launcherType);
+#if LY_HEADLESS
+        AZ::SettingsRegistryMergeUtils::MergeSettingsToRegistry_AddSpecialization(*settingsRegistry, "headless");
+        gameApplication.SetHeadless(true);
+#else
+        gameApplication.SetHeadless(false);
+#endif // LY_HEADLESS
```

Comment:
A compile time check isn't needed here and instead the IsHeadless function should be used to query the Headless server
```suggestion
        gameApplication.SetHeadless(IsHeadless());
        if (IsHeadless())
        {
            AZ::SettingsRegistryMergeUtils::MergeSettingsToRegistry_AddSpecialization(*settingsRegistry, "headless");
        }
```

Reply:
Actually `IsHeadless` was removed since there is a `.SetHeadless(..)` method anyways, so this is still needed.